kasa/smart/modules/contact.py

```diff
@@ -0,0 +1,41 @@
+"""Implementation of contact sensor module."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ...feature import Feature
+from ..smartmodule import SmartModule
+
+if TYPE_CHECKING:
+    from ..smartdevice import SmartDevice
+
+
+class ContactSensor(SmartModule):
+    """Implementation of contact sensor module."""
+
+    REQUIRED_COMPONENT = None  # we depend on availability of key
+    REQUIRED_KEY_ON_PARENT = "open"
```

Comment:
For future proofing maybe this should be a list of keys.
```suggestion
    REQUIRED_KEYS_ON_PARENT = ["open"]
```

Reply:
Given how easy it would be to change this if needed, I think it doesn't make to complicate it (yet) unnecessarily.